integration/util.py

```diff
@@ -277,6 +282,17 @@ def leave(self, folder_name):
         """
         magic-folder leave
         """
+        if self._global_config is not None:
+            # If we've accessed the folder state database from the integration
+            # tests, make sure that the connection has been closed before we
+            # try to remove the database. This is necessary on windows,
+            # otherwise the state database can't be removed.
```

Comment:
It seems like this might suggest that we actually shouldn't even keep a long-lived connection to the database?  Perhaps there are other factors to balance this consideration against ... but "connecting" to a SQLite3 database is not a particularly expensive operations and simplifying resource management by cutting down on the number of long-lived resources to manage might be worth it.

Reply:
I don't think it makes sense for each test that access a folder state database to reach into `MagicFolderConfig.database` and close it (I suspect that attribute should actually be private).

It isn't so much that we want to keep long-lived connections open, but that there isn't a good API for closing them. Part of the issue is we keep a cache of `MagicFolderConfig` obejcts on `GlobalConfigDatabase` (to ensure that we only have a single connection to each state database, and that we have access to it to close it), and there is not an API for removing an object from that cache. (See https://github.com/LeastAuthority/magic-folder/pull/475/commits/f4bbe95402dbddb2b052e77ab29042d2ad4b85ca for details.)  I'm open to other suggestions on how to handle the interactions here.